packages/react-native/android/src/main/java/backtrace/library/BacktraceDirectoryProvider.java

```diff
@@ -0,0 +1,85 @@
+package backtrace.library;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.WritableArray;
+import com.facebook.react.bridge.WritableNativeArray;
+import com.facebook.react.module.annotations.ReactModule;
+
+import java.io.File;
+import java.util.stream.Stream;
+
+
+@ReactModule(name = BacktraceDirectoryProvider.NAME)
+public class BacktraceDirectoryProvider extends ReactContextBaseJavaModule {
+    public static final String NAME = "BacktraceDirectoryProvider";
+
+    private final Context context;
+
+    public BacktraceDirectoryProvider(ReactApplicationContext reactContext) {
+        super(reactContext);
+        this.context = reactContext.getApplicationContext();
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return NAME;
+    }
+
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    public WritableArray readDirSync(String path) {
+        File file = new File(path);
+
+        WritableArray array = new WritableNativeArray();
+        for (File directoryFile : file.listFiles()) {
+            array.pushString(directoryFile.getPath());
+        }
+
+        return array;
+    }
+
+    @ReactMethod
+    public void readDir(String path, Promise promise) {
+        File file = new File(path);
+
+        WritableArray array = new WritableNativeArray();
+        for (File directoryFile : file.listFiles()) {
+            array.pushString(directoryFile.getName());
+        }
+
+        promise.resolve(array);
+    }
+
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    public boolean createDirSync(String path) {
+        File directory = new File(path);
+
+        if (directory.exists()) {
+            return true;
+        }
+        return directory.mkdirs();
+    }
+
+    @ReactMethod
+    public void createDir(String path, Promise promise) {
+        File directory = new File(path);
+
+        if (directory.exists()) {
+            promise.resolve(true);
+            return;
+        }
+        promise.resolve(directory.mkdirs());
```

Comment:
mkdris is an equivalent of the `mkdir -p `

however if the dir already exists, it returns false. Which is not something we expect to return from the method. It is just better to return true to not try it once again or to avoid disabling the database.

Here is an implementation:

```
  public boolean mkdirs() {
        if (exists()) {
            return false;
        }
        if (mkdir()) {
            return true;
        }
        File canonFile = null;
        try {
            canonFile = getCanonicalFile();
        } catch (IOException e) {
            return false;
        }

....
```

Reply:
So let's just always return `true`. It doesn't make sense to check if the directory exists twice.

Reply:
If the directory cannot be created for any kind of purpose, we cannot start crash handler and other things we're doing in the database. If the method returns `false` I still need to verify what is the reason of `false`. Calling it earlier is faster and easier to understand.